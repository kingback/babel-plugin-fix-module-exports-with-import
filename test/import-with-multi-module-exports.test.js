const babel = require('babel-core');
const plugin = require('../index');

var example = `
import { A } from 'a';
import B from 'b';
// ...
if (B) {
  module.exports = B;
} else {
  module['exports'] = A;
}
`;

it('import-with-multi-module-exports', () => {
  const {code} = babel.transform(example, {plugins: [plugin]});
  expect(code).toMatchSnapshot();
});