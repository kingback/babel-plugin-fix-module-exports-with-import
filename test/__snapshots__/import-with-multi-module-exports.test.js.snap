// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`import-with-multi-module-exports 1`] = `
"var __MIX_IMPORT_MODULE_EXPORTS__;

import { A } from 'a';
import B from 'b';
// ...
if (B) {
  __MIX_IMPORT_MODULE_EXPORTS__ = B;
} else {
  __MIX_IMPORT_MODULE_EXPORTS__ = A;
}
export default __MIX_IMPORT_MODULE_EXPORTS__;"
`;
